src/html/HTMLWriter.jl

```diff
@@ -745,6 +745,79 @@ const _SEARCHRECORD_IGNORED_BLOCK_TYPES = Union{
     Documenter.DocsNodesBlock,
     Documenter.SetupNode,
 }
+
+# Content segment for grouping content by sections
+struct ContentSegment
+    section_header::Union{Node, Nothing}  # AnchoredHeader node or nothing for default section
+    content_nodes::Vector{Node}           # Content nodes belonging to this section
+end
+
+# Segment page content by sections for improved search indexing
+function segment_page_by_sections(page_mdast::Node)
+    segments = ContentSegment[]
+    current_segment_content = Node[]
+    current_section_header = nothing
```

Comment:
How about using the type `ContentSegment` to make this code a bit more compact?
```suggestion
    current = ContentSegment(nothing, Node[])
```